tools/nodejs_api/src_cpp/node_connection.cpp

```diff
@@ -0,0 +1,119 @@
+#include "node_connection.h"
+#include "node_database.h"
+
+#include "main/kuzu.h"
+
+using namespace kuzu::main;
+
+Napi::FunctionReference NodeConnection::constructor;
+
+Napi::Object NodeConnection::Init(Napi::Env env, Napi::Object exports) {
+  Napi::HandleScope scope(env);
+
+  Napi::Function t = DefineClass(env, "NodeConnection", {
+      InstanceMethod("execute", &NodeConnection::Execute),
+      InstanceMethod("setMaxNumThreadForExec", &NodeConnection::SetMaxNumThreadForExec),
+      InstanceMethod("getNodePropertyNames", &NodeConnection::GetNodePropertyNames),
+  });
+
+  constructor = Napi::Persistent(t);
+  constructor.SuppressDestruct();
+
+  exports.Set("NodeConnection", t);
+  return exports;
+}
+
+
+NodeConnection::NodeConnection(const Napi::CallbackInfo& info) : Napi::ObjectWrap<NodeConnection>(info)  {
+  Napi::Env env = info.Env();
+  Napi::HandleScope scope(env);
+
+  if (info.Length()!=2 || !info[0].IsObject() || !info[1].IsNumber()) {
```

Comment:
the error msg is confusing given the three check in this if statement

Reply:
It only says` needs database class passed in` because on the javascript side numThreads (the second param) has a default value so the user doesn't need to specify it - hence an error will only occur if a valid database class isn't passed in.